src/main/kotlin/com/pubnub/api/retry/RetryConfiguration.kt

```diff
@@ -0,0 +1,109 @@
+package com.pubnub.api.retry
+
+import org.slf4j.LoggerFactory
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+private const val MIN_DELAY = 2
+private const val MAX_DELAY = 150
+private const val MAX_RETRIES_IN_LINEAR = 10
+private const val MAX_RETRIES_IN_EXPONENTIAL = 6
+
+/**
+ * This sealed class represents the various retry policies for a request.
+ */
+sealed class RetryConfiguration {
+
+    /**
+     * None represents no retry policy in a network request
+     */
+    object None : RetryConfiguration()
+
+    /**
+     * This data class represents a linear retry policy for network requests with a delay between retries,
+     * a maximum number of retries and a list of operations to exclude from retries.
+     *
+     * @property delayInSec The delay in seconds between retries. Minimum value is 3 seconds.
+     * @property maxRetryNumber The maximum number of retries allowed. Maximum value is 10.
+     * @property excludedOperations List of [RetryableEndpointGroup] to be excluded from retry.
+     */
+    class Linear internal constructor(
+        var delayInSec: Duration = MIN_DELAY.seconds,
```

Comment:
I see your confusion. I also had that. The thing is that in public facing constructor we have :

```
        constructor(
            minDelayInSec: Int = MIN_DELAY, // min value is 2
            maxDelayInSec: Int = MAX_DELAY, // max value is 150
            maxRetryNumber: Int = MAX_RETRIES_IN_EXPONENTIAL, // max value is 6
            excludedOperations: List<RetryableEndpointGroup> = emptyList(),
        ) : this(minDelayInSec.seconds, maxDelayInSec.seconds, maxRetryNumber, excludedOperations, false)
```
so the user will fill minDelayInSec and maxDelayInSec as Integer, in seconds.

Confusing part in that in our internal constructor we call those values like minDelayInSec/maxDelayInSec even they are of type Duration.

Renaming internal constructor like this:

```
    class Exponential internal constructor(
        var **minDelay**: Duration = MIN_DELAY.seconds, // min value is 2
        var **maxDelay**: Duration = MAX_DELAY.seconds, // max value is 150
        var maxRetryNumber: Int = MAX_RETRIES_IN_EXPONENTIAL, // max value is 6
        val excludedOperations: List<RetryableEndpointGroup> = emptyList(),
        isInternal: Boolean = false
    ) : RetryConfiguration() {
```
will cause an issue with KDocs e.g. for
`* @property minDelayInSec The minimum delay in seconds between retries. Minimum value is 3 seconds.`
there will be an error: "Cannot resolve symbol 'minDelayInSec'"

So I will leave as is. You also mentioned that "It's not super important though"

Reply:
yeah that's fine. For the future, regarding the KDoc error: that's because you're documenting a property so it has to exist on the class (as a `var` or `val`), for documenting secondary constructor **parameters**, you could do something like this:

```
/**
 * @param delayInSec something something
 * @param maxRetryNumber something smething
 */
constructor(
    delayInSec: Int = MIN_DELAY, // min value is 2
    maxRetryNumber: Int = MAX_RETRIES_IN_LINEAR, // max value is 10
    excludedOperations: List<RetryableEndpointGroup> = emptyList(),
) : this(delayInSec.seconds, maxRetryNumber, excludedOperations, false)
```